scapy/layers/l2.py

```diff
@@ -117,8 +117,14 @@ class SourceMACField(MACField):
 
     def __init__(self, name, getif=None):
         MACField.__init__(self, name, None)
-        self.getif = ((lambda pkt: pkt.payload.route()[0])
-                      if getif is None else getif)
+        if getif is None:
+            def getif(pkt):
+                try:
+                    return pkt.payload.route()[0]
+                except:
```

Comment:
Is it possible to catch a specific exception here? Otherwise please catch Exception